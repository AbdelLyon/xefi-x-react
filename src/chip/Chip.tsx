import { Chip as ChipRoot, type ChipProps } from "@heroui/react";
import type { JSX } from "react";
import { forwardRef } from "react";

export const Chip = forwardRef<HTMLDivElement, ChipProps>(
  (
    {
      children,
      avatar,
      startContent,
      endContent,
      variant = "solid",
      color = "default",
      size = "md",
      radius = "full",
      isDisabled = false,
      className,
      classNames,
      onClose,
      ...props
    },
    ref,
  ): JSX.Element => {
    return (
      <ChipRoot
        ref={ref}
        avatar={avatar}
        startContent={startContent}
        endContent={endContent}
        variant={variant}
        color={color}
        size={size}
        radius={radius}
        isDisabled={isDisabled}
        className={className}
        classNames={classNames}
        onClose={onClose}
        {...props}
      >
        {children}
      </ChipRoot>
    );
  },
);

Chip.displayName = "Chip";
