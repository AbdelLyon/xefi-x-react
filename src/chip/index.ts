import "@/index.css";
export { Chip } from "@/chip/Chip";
