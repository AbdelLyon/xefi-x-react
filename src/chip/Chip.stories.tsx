import type { Meta, StoryObj } from '@storybook/react-vite';
import {
  IconTag,
  IconX,
  IconCheck,
  IconStar,
  IconHeart,
  IconUser,
  IconHome,
  IconSettings,
  IconBell,
  IconMail,
  IconPhone,
  IconCamera,
  IconMusic,
  IconVideo,
  IconGamepad,
  IconBook,
} from '@tabler/icons-react';
import { Chip, ChipGroup } from './Chip';
import { Avatar } from '../avatar/Avatar';

const meta: Meta<typeof Chip> = {
  title: 'Components/Chip',
  component: Chip,
  parameters: {
    layout: 'centered',
    docs: {
      description: {
        component:
          '**Chip** - Composant chip avancé avec sélection, animation et validation. Support des groupes et interactions personnalisées.',
      },
    },
  },
  argTypes: {
    children: {
      control: 'text',
      description: 'Contenu du chip',
    },
    variant: {
      control: 'select',
      options: ['solid', 'bordered', 'light', 'flat', 'faded', 'shadow', 'dot'],
      description: 'Style du chip',
    },
    color: {
      control: 'select',
      options: ['default', 'primary', 'secondary', 'success', 'warning', 'danger'],
      description: 'Couleur thématique',
    },
    size: {
      control: 'select',
      options: ['sm', 'md', 'lg'],
      description: 'Taille du chip',
    },
    selectable: {
      control: 'boolean',
      description: 'Chip sélectionnable',
    },
    isSelected: {
      control: 'boolean',
      description: 'État sélectionné',
    },
    animated: {
      control: 'boolean',
      description: 'Animations activées',
    },
    onClose: {
      action: 'closed',
      description: 'Handler de fermeture',
    },
    onClick: {
      action: 'clicked',
      description: 'Handler de clic',
    },
    tooltip: {
      control: 'text',
      description: 'Texte du tooltip',
    },
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

// Story par défaut
export const Default: Story = {
  args: {
    children: 'Chip par défaut',
    color: 'primary',
    variant: 'solid',
  },
};

// Variantes de style
export const Variants: Story = {
  render: () => (
    <div className="space-y-6">
      {(['solid', 'bordered', 'light', 'flat', 'faded', 'shadow', 'dot'] as const).map((variant) => (
        <div key={variant} className="space-y-2">
          <h4 className="text-sm font-semibold capitalize">{variant}</h4>
          <div className="flex gap-2 flex-wrap">
            {(['primary', 'secondary', 'success', 'warning', 'danger'] as const).map((color) => (
              <Chip key={color} color={color} variant={variant}>
                {color}
              </Chip>
            ))}
          </div>
        </div>
      ))}
    </div>
  ),
};

// Couleurs disponibles
export const Colors: Story = {
  render: () => (
    <div className="grid grid-cols-3 gap-4">
      {(['default', 'primary', 'secondary', 'success', 'warning', 'danger'] as const).map((color) => (
        <div key={color} className="text-center space-y-2">
          <Chip color={color}>{color}</Chip>
          <p className="text-xs text-default-500 capitalize">{color}</p>
        </div>
      ))}
    </div>
  ),
};

// Tailles
export const Sizes: Story = {
  render: () => (
    <div className="flex items-center gap-4">
      <div className="text-center">
        <Chip size="sm" color="primary">Small</Chip>
        <p className="text-xs mt-2 text-default-500">Small</p>
      </div>
      <div className="text-center">
        <Chip size="md" color="primary">Medium</Chip>
        <p className="text-xs mt-2 text-default-500">Medium</p>
      </div>
      <div className="text-center">
        <Chip size="lg" color="primary">Large</Chip>
        <p className="text-xs mt-2 text-default-500">Large</p>
      </div>
    </div>
  ),
};

// Chips avec icônes
export const WithIcons: Story = {
  render: () => (
    <div className="space-y-6">
      <div>
        <h4 className="text-sm font-semibold mb-3">Icônes de début</h4>
        <div className="flex gap-2 flex-wrap">
          <Chip
            color="primary"
            startContent={<IconUser size={16} />}
          >
            Utilisateur
          </Chip>
          <Chip
            color="success"
            startContent={<IconCheck size={16} />}
          >
            Validé
          </Chip>
          <Chip
            color="warning"
            startContent={<IconStar size={16} />}
          >
            Favori
          </Chip>
          <Chip
            color="danger"
            startContent={<IconHeart size={16} />}
          >
            Aimé
          </Chip>
        </div>
      </div>

      <div>
        <h4 className="text-sm font-semibold mb-3">Icônes de fin</h4>
        <div className="flex gap-2 flex-wrap">
          <Chip
            color="primary"
            endContent={<IconSettings size={16} />}
          >
            Configuration
          </Chip>
          <Chip
            color="secondary"
            endContent={<IconBell size={16} />}
          >
            Notifications
          </Chip>
        </div>
      </div>

      <div>
        <h4 className="text-sm font-semibold mb-3">Les deux</h4>
        <div className="flex gap-2 flex-wrap">
          <Chip
            color="success"
            startContent={<IconHome size={16} />}
            endContent={<IconCheck size={16} />}
          >
            Accueil vérifié
          </Chip>
        </div>
      </div>
    </div>
  ),
};

// Chips avec avatars
export const WithAvatars: Story = {
  render: () => (
    <div className="flex gap-2 flex-wrap">
      <Chip
        variant="flat"
        avatar={
          <Avatar
            name="JD"
            size="sm"
            src="https://i.pravatar.cc/150?u=a042581f4e29026024d"
          />
        }
      >
        John Doe
      </Chip>
      <Chip
        variant="flat"
        color="secondary"
        avatar={
          <Avatar
            name="JS"
            size="sm"
            src="https://i.pravatar.cc/150?u=a04258a2462d826712d"
          />
        }
      >
        Jane Smith
      </Chip>
      <Chip
        variant="flat"
        color="success"
        avatar={<Avatar name="BW" size="sm" color="success" />}
      >
        Bob Wilson
      </Chip>
    </div>
  ),
};

// Chips sélectionnables
export const Selectable: Story = {
  render: () => (
    <div className="space-y-6">
      <div>
        <h4 className="text-sm font-semibold mb-3">Sélection simple</h4>
        <div className="flex gap-2 flex-wrap">
          <Chip
            selectable
            color="primary"
            startContent={<IconTag size={16} />}
          >
            React
          </Chip>
          <Chip
            selectable
            color="secondary"
            startContent={<IconTag size={16} />}
          >
            TypeScript
          </Chip>
          <Chip
            selectable
            color="success"
            startContent={<IconTag size={16} />}
          >
            Vite
          </Chip>
          <Chip
            selectable
            color="warning"
            startContent={<IconTag size={16} />}
          >
            Storybook
          </Chip>
        </div>
        <p className="text-xs text-default-500 mt-2">Cliquez pour sélectionner/désélectionner</p>
      </div>

      <div>
        <h4 className="text-sm font-semibold mb-3">Pré-sélectionnés</h4>
        <div className="flex gap-2 flex-wrap">
          <Chip
            selectable
            isSelected
            color="primary"
            variant="bordered"
          >
            Frontend
          </Chip>
          <Chip
            selectable
            isSelected
            color="success"
            variant="bordered"
          >
            Backend
          </Chip>
          <Chip
            selectable
            color="secondary"
            variant="bordered"
          >
            Mobile
          </Chip>
        </div>
      </div>
    </div>
  ),
};

// Chips avec fermeture
export const Closable: Story = {
  render: () => (
    <div className="space-y-6">
      <div>
        <h4 className="text-sm font-semibold mb-3">Chips fermables</h4>
        <div className="flex gap-2 flex-wrap">
          <Chip
            color="primary"
            onClose={() => console.log('Fermé: JavaScript')}
          >
            JavaScript
          </Chip>
          <Chip
            color="secondary"
            onClose={() => console.log('Fermé: Python')}
          >
            Python
          </Chip>
          <Chip
            color="success"
            onClose={() => console.log('Fermé: Java')}
          >
            Java
          </Chip>
          <Chip
            color="warning"
            onClose={() => console.log('Fermé: C++')}
          >
            C++
          </Chip>
        </div>
        <p className="text-xs text-default-500 mt-2">Cliquez sur le X pour fermer</p>
      </div>

      <div>
        <h4 className="text-sm font-semibold mb-3">Avec avatars et fermeture</h4>
        <div className="flex gap-2 flex-wrap">
          <Chip
            variant="flat"
            avatar={<Avatar name="AL" size="sm" color="primary" />}
            onClose={() => console.log('Retiré: Alice')}
          >
            Alice Martin
          </Chip>
          <Chip
            variant="flat"
            color="secondary"
            avatar={<Avatar name="BJ" size="sm" color="secondary" />}
            onClose={() => console.log('Retiré: Bob')}
          >
            Bob Johnson
          </Chip>
        </div>
      </div>
    </div>
  ),
};

// Chips avec tooltips
export const WithTooltips: Story = {
  render: () => (
    <div className="flex gap-2 flex-wrap">
      <Chip
        color="primary"
        tooltip="Framework JavaScript moderne"
        startContent={<IconTag size={16} />}
      >
        React
      </Chip>
      <Chip
        color="secondary"
        tooltip="Superset typé de JavaScript"
        startContent={<IconTag size={16} />}
      >
        TypeScript
      </Chip>
      <Chip
        color="success"
        tooltip="Build tool rapide"
        startContent={<IconTag size={16} />}
      >
        Vite
      </Chip>
    </div>
  ),
};

// ChipGroup - Groupes de chips
export const ChipGroups: Story = {
  render: () => (
    <div className="space-y-8 max-w-4xl">
      <div>
        <h4 className="text-sm font-semibold mb-4">Groupe basique</h4>
        <ChipGroup
          chips={[
            { children: 'React', color: 'primary', startContent: <IconTag size={16} /> },
            { children: 'Vue', color: 'success', startContent: <IconTag size={16} /> },
            { children: 'Angular', color: 'danger', startContent: <IconTag size={16} /> },
            { children: 'Svelte', color: 'warning', startContent: <IconTag size={16} /> },
          ]}
        />
      </div>

      <div>
        <h4 className="text-sm font-semibold mb-4">Sélection multiple</h4>
        <ChipGroup
          selectable
          selectionMode="multiple"
          chips={[
            { children: 'HTML', color: 'primary', key: 'html' },
            { children: 'CSS', color: 'secondary', key: 'css' },
            { children: 'JavaScript', color: 'warning', key: 'js' },
            { children: 'TypeScript', color: 'success', key: 'ts' },
            { children: 'React', color: 'primary', key: 'react' },
            { children: 'Node.js', color: 'success', key: 'node' },
          ]}
          wrap
        />
        <p className="text-xs text-default-500 mt-2">Sélection multiple autorisée</p>
      </div>

      <div>
        <h4 className="text-sm font-semibold mb-4">Sélection unique</h4>
        <ChipGroup
          selectable
          selectionMode="single"
          chips={[
            { children: 'Facile', color: 'success', key: 'easy' },
            { children: 'Moyen', color: 'warning', key: 'medium' },
            { children: 'Difficile', color: 'danger', key: 'hard' },
            { children: 'Expert', color: 'secondary', key: 'expert' },
          ]}
        />
        <p className="text-xs text-default-500 mt-2">Un seul choix possible</p>
      </div>

      <div>
        <h4 className="text-sm font-semibold mb-4">Orientations et espacements</h4>
        <div className="space-y-4">
          <div>
            <p className="text-xs text-default-500 mb-2">Horizontal - Espacement large</p>
            <ChipGroup
              orientation="horizontal"
              spacing="lg"
              chips={[
                { children: 'Large', color: 'primary' },
                { children: 'Spacing', color: 'secondary' },
                { children: 'Horizontal', color: 'success' },
              ]}
            />
          </div>
          
          <div>
            <p className="text-xs text-default-500 mb-2">Vertical</p>
            <ChipGroup
              orientation="vertical"
              spacing="md"
              chips={[
                { children: 'Vertical', color: 'primary' },
                { children: 'Layout', color: 'secondary' },
                { children: 'Mode', color: 'success' },
              ]}
            />
          </div>
        </div>
      </div>
    </div>
  ),
};

// Showcase d'utilisation pratique
export const PracticalShowcase: Story = {
  render: () => (
    <div className="max-w-6xl mx-auto p-8 space-y-8">
      <div className="mb-8">
        <h2 className="text-3xl font-bold text-foreground mb-2">
          Chips en Action
        </h2>
        <p className="text-default-500">
          Exemples d'utilisation dans des interfaces réelles
        </p>
      </div>

      {/* Filtres de recherche */}
      <div className="p-6 bg-content1 rounded-lg">
        <h3 className="text-lg font-semibold mb-4">Filtres de Recherche</h3>
        <div className="space-y-4">
          <div>
            <p className="text-sm text-default-600 mb-2">Catégories:</p>
            <ChipGroup
              selectable
              selectionMode="multiple"
              spacing="sm"
              chips={[
                { children: 'Électronique', color: 'primary', startContent: <IconCamera size={14} /> },
                { children: 'Mode', color: 'secondary', startContent: <IconHeart size={14} /> },
                { children: 'Maison', color: 'success', startContent: <IconHome size={14} /> },
                { children: 'Sport', color: 'warning', startContent: <IconGamepad size={14} /> },
                { children: 'Livres', color: 'danger', startContent: <IconBook size={14} /> },
              ]}
            />
          </div>
          
          <div>
            <p className="text-sm text-default-600 mb-2">Prix:</p>
            <ChipGroup
              selectable
              selectionMode="single"
              spacing="sm"
              chips={[
                { children: '< 50€', color: 'success', variant: 'bordered' },
                { children: '50€ - 100€', color: 'warning', variant: 'bordered' },
                { children: '100€ - 200€', color: 'danger', variant: 'bordered' },
                { children: '> 200€', color: 'secondary', variant: 'bordered' },
              ]}
            />
          </div>
        </div>
      </div>

      {/* Tags d'article */}
      <div className="p-6 bg-content1 rounded-lg">
        <h3 className="text-lg font-semibold mb-4">Tags d'Article</h3>
        <div className="space-y-2">
          <div className="flex gap-2 flex-wrap">
            <Chip size="sm" color="primary" variant="flat">React</Chip>
            <Chip size="sm" color="secondary" variant="flat">TypeScript</Chip>
            <Chip size="sm" color="success" variant="flat">Tutorial</Chip>
            <Chip size="sm" color="warning" variant="flat">Débutant</Chip>
            <Chip size="sm" color="danger" variant="flat">Populaire</Chip>
          </div>
          <p className="text-xs text-default-500">
            Tags automatiques basés sur le contenu
          </p>
        </div>
      </div>

      {/* Équipe projet */}
      <div className="p-6 bg-content1 rounded-lg">
        <h3 className="text-lg font-semibold mb-4">Équipe Projet</h3>
        <div className="space-y-4">
          <div>
            <p className="text-sm text-default-600 mb-2">Membres assignés:</p>
            <div className="flex gap-2 flex-wrap">
              <Chip
                variant="flat"
                avatar={<Avatar name="AM" size="sm" src="https://i.pravatar.cc/150?u=alice" />}
                onClose={() => console.log('Retiré Alice')}
              >
                Alice Martin
              </Chip>
              <Chip
                variant="flat"
                color="secondary"
                avatar={<Avatar name="BJ" size="sm" src="https://i.pravatar.cc/150?u=bob" />}
                onClose={() => console.log('Retiré Bob')}
              >
                Bob Johnson
              </Chip>
              <Chip
                variant="flat"
                color="success"
                avatar={<Avatar name="CD" size="sm" color="success" />}
              >
                Charlie Davis
              </Chip>
            </div>
          </div>

          <div>
            <p className="text-sm text-default-600 mb-2">Rôles:</p>
            <div className="flex gap-2 flex-wrap">
              <Chip color="primary" size="sm" startContent={<IconUser size={14} />}>
                Lead Dev
              </Chip>
              <Chip color="secondary" size="sm" startContent={<IconSettings size={14} />}>
                DevOps
              </Chip>
              <Chip color="success" size="sm" startContent={<IconCheck size={14} />}>
                QA
              </Chip>
            </div>
          </div>
        </div>
      </div>

      {/* Préférences utilisateur */}
      <div className="p-6 bg-content1 rounded-lg">
        <h3 className="text-lg font-semibold mb-4">Préférences</h3>
        <div className="space-y-4">
          <div>
            <p className="text-sm text-default-600 mb-2">Notifications:</p>
            <ChipGroup
              selectable
              selectionMode="multiple"
              spacing="sm"
              chips={[
                { children: 'Email', color: 'primary', startContent: <IconMail size={14} />, key: 'email' },
                { children: 'Push', color: 'success', startContent: <IconBell size={14} />, key: 'push' },
                { children: 'SMS', color: 'warning', startContent: <IconPhone size={14} />, key: 'sms' },
              ]}
              selectedKeys={new Set(['email', 'push'])}
            />
          </div>

          <div>
            <p className="text-sm text-default-600 mb-2">Centres d'intérêt:</p>
            <div className="flex gap-2 flex-wrap">
              <Chip
                selectable
                color="primary"
                size="sm"
                startContent={<IconMusic size={14} />}
              >
                Musique
              </Chip>
              <Chip
                selectable
                color="secondary"
                size="sm"
                startContent={<IconVideo size={14} />}
              >
                Vidéos
              </Chip>
              <Chip
                selectable
                color="success"
                size="sm"
                startContent={<IconGamepad size={14} />}
              >
                Gaming
              </Chip>
              <Chip
                selectable
                color="warning"
                size="sm"
                startContent={<IconBook size={14} />}
              >
                Lecture
              </Chip>
            </div>
          </div>
        </div>
      </div>
    </div>
  ),
  parameters: {
    layout: 'fullscreen',
  },
};