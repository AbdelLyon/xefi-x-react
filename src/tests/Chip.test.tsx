import { Chip } from "@/chip";
import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";

describe("Composant Chip", (): void => {
  describe("Snapshots", (): void => {
    it("devrait correspondre au snapshot avec les props par défaut", (): void => {
      const { container } = render(<Chip>Chip par défaut</Chip>);
      expect(container).toMatchSnapshot();
    });

    it("devrait correspondre au snapshot avec différentes variantes", (): void => {
      const variants = [
        "solid",
        "bordered",
        "light",
        "flat",
        "faded",
        "shadow",
      ] as const;
      variants.forEach((variant): void => {
        const { container } = render(
          <Chip variant={variant}>Chip {variant}</Chip>,
        );
        expect(container).toMatchSnapshot();
      });
    });

    it("devrait correspondre au snapshot avec différentes couleurs", (): void => {
      const colors = [
        "default",
        "primary",
        "secondary",
        "success",
        "warning",
        "danger",
      ] as const;
      colors.forEach((color): void => {
        const { container } = render(<Chip color={color}>Chip {color}</Chip>);
        expect(container).toMatchSnapshot();
      });
    });

    it("devrait correspondre au snapshot avec différentes tailles", (): void => {
      const sizes = ["sm", "md", "lg"] as const;
      sizes.forEach((size): void => {
        const { container } = render(<Chip size={size}>Chip {size}</Chip>);
        expect(container).toMatchSnapshot();
      });
    });

    it("devrait correspondre au snapshot avec contenu de début et de fin", (): void => {
      const { container } = render(
        <Chip startContent={<span>Début</span>} endContent={<span>Fin</span>}>
          Contenu
        </Chip>,
      );
      expect(container).toMatchSnapshot();
    });

    it("devrait correspondre au snapshot quand désactivé", (): void => {
      const { container } = render(<Chip isDisabled>Chip désactivé</Chip>);
      expect(container).toMatchSnapshot();
    });
  });

  describe("Comportement", (): void => {
    it("devrait appeler onClose lorsque le bouton de fermeture est cliqué", (): void => {
      const onCloseMock = vi.fn();
      render(<Chip onClose={onCloseMock}>Chip fermable</Chip>);

      const closeButton = screen.getByRole("button");
      fireEvent.click(closeButton);

      expect(onCloseMock).toHaveBeenCalledTimes(1);
    });

    it("devrait transmettre les props supplémentaires au composant racine", (): void => {
      const { container } = render(
        <Chip data-testid="test-chip">Chip de test</Chip>,
      );
      expect(container.firstChild).toHaveAttribute("data-testid", "test-chip");
    });
  });

  describe("Accessibilité", (): void => {
    it("ne devrait pas avoir d'attribut role par défaut", (): void => {
      render(<Chip>Chip accessible</Chip>);
      const chip = screen.getByText("Chip accessible");
      expect(chip).not.toHaveAttribute("role");
    });

    it("ne devrait pas avoir d'attribut tabindex par défaut", (): void => {
      render(<Chip>Chip focusable</Chip>);
      const chip = screen.getByText("Chip focusable");
      expect(chip).not.toHaveAttribute("tabindex");
    });

    it("ne devrait pas être focusable quand il est désactivé", (): void => {
      render(<Chip isDisabled>Chip désactivé</Chip>);
      const chip = screen.getByText("Chip désactivé");
      expect(chip).not.toHaveAttribute("tabindex");
    });
  });
});
