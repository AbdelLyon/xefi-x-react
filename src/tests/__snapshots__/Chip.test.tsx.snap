// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec contenu de début et de fin 1`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full bg-default text-default-foreground"
  >
    <span
      class="max-h-[80%]"
    >
      Début
    </span>
    <span
      class="flex-1 text-inherit font-normal px-2 pl-1 pr-1"
    >
      Contenu
    </span>
    <span
      class="max-h-[80%]"
    >
      Fin
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes couleurs 1`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full bg-default text-default-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip 
      default
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes couleurs 2`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full bg-primary text-primary-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip 
      primary
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes couleurs 3`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full bg-secondary text-secondary-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip 
      secondary
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes couleurs 4`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full bg-success text-success-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip 
      success
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes couleurs 5`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full bg-warning text-warning-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip 
      warning
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes couleurs 6`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full bg-danger text-danger-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip 
      danger
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes tailles 1`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-6 text-tiny rounded-full bg-default text-default-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-1"
    >
      Chip 
      sm
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes tailles 2`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full bg-default text-default-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip 
      md
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes tailles 3`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-2 h-8 text-medium rounded-full bg-default text-default-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip 
      lg
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes variantes 1`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full bg-default text-default-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip 
      solid
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes variantes 2`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap border-medium px-1 h-7 text-small rounded-full bg-transparent border-default text-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip 
      bordered
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes variantes 3`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full bg-transparent text-default-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip 
      light
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes variantes 4`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full bg-default/40 text-default-700"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip 
      flat
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes variantes 5`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap border-medium px-1 h-7 text-small rounded-full border-default bg-default-100 text-default-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip 
      faded
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec différentes variantes 6`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full shadow-lg shadow-default/50 bg-default text-default-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip 
      shadow
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot avec les props par défaut 1`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full bg-default text-default-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip par défaut
    </span>
  </div>
</div>
`;

exports[`Composant Chip > Snapshots > devrait correspondre au snapshot quand désactivé 1`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full opacity-disabled pointer-events-none bg-default text-default-foreground"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      Chip désactivé
    </span>
  </div>
</div>
`;
